src/transform/src/reduce_elision.rs

```diff
@@ -94,6 +94,30 @@ impl ReduceElision {
                             exprs: vec![a.expr.clone()],
                         },
 
+                        // StringAgg takes nested records of strings and outputs a string
+                        AggregateFunc::StringAgg { .. } => {
+                            if let MirScalarExpr::CallVariadic {
+                                func: VariadicFunc::RecordCreate { .. },
+                                ref exprs,
+                            } = a.expr
+                            {
+                                if let MirScalarExpr::CallVariadic {
+                                    func: VariadicFunc::RecordCreate { .. },
+                                    ref exprs,
+                                } = exprs[0]
+                                {
+                                    exprs[0].clone()
+                                } else {
+                                    panic!(
+                                        "need nested RecordCreate as expr for StringAgg {:?}",
```

Comment:
I know `panic!` came from the reference I originally sent you, but can you replace panics with throwing a TransformError so Materialize does not crash?
Here's an example of a TransformError:
https://github.com/MaterializeInc/materialize/blob/6777438a2d81dada015efa3db96c6d9a4398344d/src/transform/src/lib.rs#L153
Here is an example of a transform that could throw an error:
https://github.com/MaterializeInc/materialize/blob/a5d980709c12e6f85a4742670ce5510f9fd73249/src/transform/src/column_knowledge.rs#L32

Reply:
sure!